### ISSUE TITLE
Writing a hashtag turns into HUGE heading

### ISSUE THREAD

**Issue by Fryguy** [gitterHQ/marked#9]
This is a request to detect a one-liner lead with a `#` without a space, as a hashtag, and to not render it with markdown.

---

Technically, it's not a bug, as the markdown processor sees text like 

```markdown
#winning
```

as a header, and turns it into 

# winning

---

However, most people don't write a markdown header into the chat with nothing else.  When they are typing like that they intend for it to be a silly comment like a hashtag, but instead get huge text rendered and are very confused.

Additionally, a typical way to do a markdown header is to put a space after the `#`, so the markdown processor is being a bit lenient in this regard.  (Aside, even GitHub's markdown processor doesn't recognize the `#` without a space after it as a header, so even to type it for demonstration above, I had to add a space)

A colleague mentioned escaping to me, but, IMO a user shouldn't have to know that (and newbies have no idea why it happens, let alone how to fix it).

---

Perhaps, extending on this request, it could even just be turned into a link to Twitter. For example:

[#winning](https://twitter.com/hashtag/winning?src=hash)

---

Moved here from https://github.com/gitterHQ/gitter/issues/1714

**Comment by MadLittleMods** [comment]
> I could see doing this for single-line messages without a space between the hash and text.
>
> @MadLittleMods, https://github.com/gitterHQ/gitter/issues/1714#issuecomment-304377746

**Comment by cben** [comment]
👍 CommonMark spec says "The opening sequence of # characters must be followed by a space or by the end of line."  (http://spec.commonmark.org/0.26/#atx-heading).  As does GFM that's now derived from CommonMark.

Babelmark shows no consensus: https://johnmacfarlane.net/babelmark2/?normalize=1&text=%23header-or-hashtag

CommonMark discussion: https://talk.commonmark.org/t/heading-not-working/819/26
TL;DR: requiring space was deliberate, clearly a win for short-form use cases like chat, debatable for say blog posts but only due to compatibility concern with existing texts using `#header`.
Notably, Github always required space to avoid clash with `#42` issue notation, and I think "least surprise" criterion for Gitter syntax should align with Github.

**Comment by Fryguy** [comment]
Note that quoting also renders a huge heading as well...

```markdown
> #winning
```

renders as

> # winning
